Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/exceptionhandler/CustomExceptionHandlerWrapper.java

```diff
@@ -58,28 +64,39 @@ public ExceptionHandler getWrapped() {
 
     @Override
     public void handle() {
-
         final Iterator<ExceptionQueuedEvent> i = getUnhandledExceptionQueuedEvents().iterator();
         while (i.hasNext()) {
             ExceptionQueuedEvent event = i.next();
             ExceptionQueuedEventContext context
                 = (ExceptionQueuedEventContext) event.getSource();
-
             // get the exception from context
             Throwable t = context.getException();
-            LOG.error("An exception occurred whil performing user request", t);
+            LOG.error("An exception occurred while performing user request", t);
             final FacesContext fc = FacesContext.getCurrentInstance();
             final Map<String, Object> requestMap = fc.getExternalContext().getRequestMap();
             final NavigationHandler nav = fc.getApplication().getNavigationHandler();
 
+            HttpServletRequest request = (HttpServletRequest) fc.getExternalContext().getRequest();
+            Object servletEx = request.getAttribute("javax.servlet.error.exception");
             try {
-
-                //redirect error page
-                FacesContext.getCurrentInstance().addMessage("errorMessage", new FacesMessage(FacesMessage.SEVERITY_ERROR,
-                    "An Exception occurred while performing the requested function.", ""));
-                FacesContext.getCurrentInstance().getExternalContext().invalidateSession();
-                requestMap.put("exceptionMessage", t.getMessage());
-                nav.handleNavigation(fc, null, "/customerror");
+                if (servletEx instanceof SanitizationException) {
+                    //SanitizationFilter has redirected to customerror page
+                    fc.addMessage("errorMessage", new FacesMessage(FacesMessage.SEVERITY_ERROR,
+                        "An Exception occurred while performing the requested function.", ""));
```

Comment:
Moved exception message  to a constant class.  Created a backlog item to introduce ResourceBundle to admingui.